### PR TITLE
Authentication Ruby on Rails: Members Only

### PR DESCRIPTION
**Pull request by janis-jenny**
What this PR does:

- Adds post model, controllers, and views

- Adds user authentication using Devise

- Adds view logic to distinguish between users who are logged in or not

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
-  </head>
 
+  </head>
   <body>
-    <%= yield %>
+    <div class= "container">
+      <%= yield %>
+    </div>
+    <!-- Optional JavaScript; choose one of the two! -->
+
```
Comment (lekegitrepo) [inline app/views/layouts/application.html.erb]: - **_The application layout should have Sign in / Sign out links (So they are not manually repeated in every view) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle:
### Hi @janis-jenny and @AlanSoto31 :clap: :clap:
#### The project is well implemented and looks great. However, some requirements are missing. Check my comments below.

#### App Behaviour:
- **_Fix the Signup button: it currently displays the login page when clicked instead of the registration/signup page_**

#### Optional:
- **_Add Flash messages to display unsuccessful actions_**

##### _Reviewed by Temitayo Adeleke_

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Great Job Team!
I noticed while you were updating the changes requested by the previous TSE, you violated a few rules. Kindly fix the outlined changes and you should be done.

- [ ] In your `posts_controller` and `views/application.html.erb` file kindly remove the comments on line 20 and 23. It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. Affected files: `views/application.html.erb and posts/index.html.erb` **The Resources below can come in handy**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user.id
```
Comment (teekaytech) [inline app/controllers/posts_controller.rb]: Here's the best way to do this
`@post = current_user.posts.build(post_params)`

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__This project is looking good. All functionalities seem to be working, except for one major bug. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] Whenever I create a new post, all existing posts' authors are changed to the author of the newly created post. Kindly review your logic and fix this.
- To see this bug, 
  - Create 2 or more users, create a post by the first user, then logout.
  - Login as the second user and create another post. 
  - Observe the posts page, as all authors will have changed to the second user.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by acushlakoncept — APPROVED** [review]
# Status: PROJECT APPROVED  🎉 🚀

Great Job on completing all the project requirements. I see you have successfully implemented all the changes requested by the previous TSEs.

I will go ahead and mark this as complete, feel free to merge when you are ready. All the best with the next project.


### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Pull request merged**